WickedEngine/wiLua.cpp

```diff
@@ -76,7 +76,8 @@ namespace wi::lua
 		std::string filepath = filename;
 		std::replace(filepath.begin(), filepath.end(), '\\', '/');
 
-		std::string dynamic_inject = "local function script_file() return \"" + filepath + "\" end;";
+		std::string dynamic_inject = "--" + filepath + "\n";
```

Comment:
I think the `\n` linebreak will be an issue, because it will offset line indicators of error messages compared to the real file, so you should instead just use `;` at the end of these injected statements.

Reply:
Ahh, I have adjusted this so that it no longer messes with line count.

Also cleaned up error handling a little bit.